### PR TITLE
feat: compact JSONC formatting (including comments) (#578)

### LINKED ISSUE #578: Spec Format
Any chance we can get the format of the JSON in the spec to match the format of JSON in the online editor when right-clicking Format Document in Deneb? Currently, it adds many more carriage returns which makes the spec very long and hard to read. I always end of formatting in the online editor and then pasting back into Deneb at the moment.

### PR DESCRIPTION
**Pull request by dm-p**
## What this does

Fixes #578. Replaces the verbose one-value-per-line JSON formatting (jsonc-parser `format()`) with a **comment-preserving compact formatter** that packs objects and arrays onto a single line when they fit — the same layout Vega Editor produces via `json-stringify-pretty-compact` — while keeping full JSONC comment support.

```jsonc
// before                                   // after
{                                           {
  "mark": {                                   "mark": {"type": "bar", "tooltip": true},
    "type": "bar",                            "encoding": {
    "tooltip": true                             "x": {"field": "Category", "type": "nominal"},
  },                                            "y": {"field": "Sales", "type": "quantitative"}
  ...                                         }
}                                           }
```

## Changes

- **New formatter** (`@deneb-viz/utils/jsonc`): walks the jsonc-parser AST and applies the pretty-compact fit rule — a container stays on one line when its single-line form (indent + `"key": ` prefix + trailing comma included) fits within the max line length *and* contains no comments. Literals are copied as raw source slices, so `1.0`, `1e3`, and string escapes survive untouched. Invalid JSON is returned unchanged.
- **New property — “Maximum line length when formatting”** (Advanced editor → JSON editor): 40–200, default 80 (Vega Editor's effective default). Changes apply to the next format without reloading.
- **Monaco integration**: the JSON worker's built-in formatter is disabled and replaced with document + range providers backed by the new formatter. **Format Selection** now appears in the context menu, and <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>R</kbd> formats the selection when one exists; otherwise, it formats the whole document. Output adopts the document's existing line-ending style (CRLF-safe, including the invalid-JSON no-op path).
- **Template import/export** use the same formatter and honor the property, so exported templates and imported specs are as readable as the editor.

## Behavior notes (documented in DEVELOPMENT.md §5)

- Comments force expansion of their container (and every ancestor). Leading comments stay above their value; same-line trailing comments stay on the line; a comment between a key and its value is moved above the next entry.
- Range formatting snaps outward to the smallest complete value/property containing the selection.
- Layout is deterministic and idempotent — existing line breaks are not preserved.

## Testing

- **Parity oracle**: comment-free output is asserted byte-for-byte against `json-stringify-pretty-compact` (already a dependency) across Vega-Lite fixtures, edge cases, and multiple width/tab settings.
- 35 formatter unit tests (comment attachment, literal fidelity, range snapping incl. the comma-reserve fit boundary, idempotence, invalid input) + 6 Monaco provider tests exercising the *real* formatter (no mocks), including CRLF regression pins.
- Full suite green: utils 133, json-processing 102, app-core 664, root 565; `npm run ci:local` passes.
- Manual verification in Desktop: format document/selection via hotkey and context menu, live property changes, template export→import round-trip, comment preservation.

## Notes

- No new dependencies; net bundle impact ≈ 0 (the worker's unused formatter remains in the worker bundle — not tree-shakeable).
- Targeted at **2.1** — hold merge until the 2.0 certification cut is taken from `main`.
- Design record: `docs/brainstorms/2026-08-21-compact-jsonc-formatting-requirements.md`; implementation plan: `docs/plans/2026-08-21-001-compact-jsonc-formatting-plan.md`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/utils/src/lib/jsonc-format.ts [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+        siblings.length === 0 || siblings[siblings.length - 1] === target;
+    return {
+        offset: target.offset,
+        length: target.length,
+        content: renderNode(
+            ctx,
+            target,
+            countContainerAncestors(target),
+            isLast ? 0 : 1
+        )
```
[the remark this answers is not in the document]
Comment (dm-p) [reply packages/utils/src/lib/jsonc-format.ts:472]: Valid — confirmed with a reproducing test (key/value-gap comment deleted by Format Selection). Fixed in b7311598: the range target now widens to the nearest ancestor whose rendering carries every comment inside the replaced span (terminates at the root, where all in-span comments are attached within the subtree). Covers both attachment shapes (leading-of-next-sibling and inner-of-enclosing-container), with non-widening guard tests proving narrow edits stay narrow, plus a DEVELOPMENT.md note on the behaviour.
